src/movies.js

```diff
@@ -1,16 +1,92 @@
 // Iteration 1: All directors? - Get the array of all directors.
+function getAllDirectors(array){
+    let dirArray=array.map((movie)=>{
+        return movie.director;
+    })
+    return dirArray
+}
 // _Bonus_: It seems some of the directors had directed multiple movies so they will pop up multiple times in the array of directors. How could you "clean" a bit this array and make it unified (without duplicates)?
 
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct?
+function howManyMovies(array){
+    let filterArray=array.filter((movie)=>{
+        return  movie.director==="Steven Spielberg" && movie.genre.includes("Drama")
+
+    })
+    return filterArray.length
+}
 
 // Iteration 3: All rates average - Get the average of all rates with 2 decimals
+function ratesAverage(array){
+    if(array.length===0)
+    {return 0};
+    let totRate=array.reduce((acc,elem)=>{
+        return acc+elem.rate
```

Comment:
the condition to avoid the movies which do not have a rate to pass this missing test could be applied here. A simple conditional that checks if (elem.rate). Otherwise the value undefined would try to be added to the accumulator and cause the value to end up being NaN.